api/backup_endpoints.go

```diff
@@ -3,13 +3,18 @@ package api
 import (
 	"fmt"
 	"strconv"
+	"sync"
+)
+
+var (
```

Comment:
You can use the cache package here so we don't have to manually handle mutexes and it'll auto-clear after enough time to avoid a random memory sink.